### PR TITLE
[MINOR] EDA-1283 filter my bots

### PR DESCRIPTION
**Pull request by ibrizuela-eb**
- Modify the hardcoded my_bots for a method to list the real my online bots.
- I need help to test the method `get_data_users`
- The rest of the code is tested.

### CODE REVIEW

File: development/forms.py [hunk 1]
```diff
@@ -2,33 +2,49 @@
...
 SERVER_PORT = get_env_variable('SERVER_PORT')
 
-my_bots = [
-    ('0', 'Pedro'),
-]
-
 
 class ChallengeForm(forms.Form):
     def __init__(self, *args, **kwargs):
+        user = kwargs.pop('user')
```
Comment (eb-cmoyano) [inline development/forms.py:14]: Please, remove this line since you're already initiating the user in the `get_form` method.

File: development/forms.py [hunk 2]
```diff
@@ -2,33 +2,49 @@
...
+        users = []
+
+    return users
+
+
+def get_online_bots(users):
+    return enumerate(users)
+
+
+def get_my_bots(user_online, online_bots):
```
Comment (eb-cmoyano) [inline development/forms.py:46]: Should these methods live on `forms.py`? IMO, looks like helper/domain methods..

File: development/forms.py [hunk 3]
```diff
@@ -2,33 +2,49 @@
...
             '{}:{}/users'.format(SERVER_URL, SERVER_PORT),
         )
         data = bots_json.json()
-        on_bots = [(str(i), bot) for i, bot in enumerate(data['users'])]
-        return on_bots
+        users = data['users']
     except Exception:
-        on_bots = []
-        return on_bots
+        users = []
```
Comment (eb-cmoyano) [inline development/forms.py:37]: can you catch a more specific exception instead of a base Exception?

File: development/forms.py (hunk 1 above)
Comment (nicolasnunez-eb) [inline development/forms.py:14]: be careful using pop you have to consider that pop perform a side effect on the dictionary (it returns you the value but it removes that value from the dict)

File: development/tests.py [hunk 4]
```diff
@@ -42,3 +50,34 @@ def test_create_match(
...
+            token='asd123',
+            user=user,
+        )
+        online_bots = ['Pedro', 'bot_name', 'Pablo']
+        self.assertEqual(
+            list(get_my_bots(user, online_bots)),
+            [
+                (0, 'bot_name')
+            ],
+        )
```
Comment (eb-cmoyano) [inline development/tests.py:83]: You are not testing forms here. Actually, you're testing helper methods. Also, they are quite different, could you split them into different test suites?

**Review by nicolasnunez-eb — COMMENTED** [review]
You can test the get_data_users patching the requests library and mocking the different scenarios that you can have

**Review by nicolasnunez-eb — COMMENTED** [review]
You can test the get_data_users patching the requests library and mocking the different scenarios that you can have

File: development/forms.py (hunk 2 above)
Comment (ibrizuela-eb) [reply development/forms.py]: done!

File: development/forms.py (hunk 1 above)
Comment (ibrizuela-eb) [reply development/forms.py]: Fixed!

File: development/bot_handler.py [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+    try:
+        bots_json = requests.get(
+            '{}:{}/users'.format(SERVER_URL, SERVER_PORT),
+        )
+        data = bots_json.json()
+        users = data['users']
+    except Exception:
+        users = []
+
+    return users
```
Comment (nicolasnunez-eb) [inline development/bot_handler.py:21]: nit: you can directly return without creating a temp variable like
```
try:
      ....
      return data['users']
except Exception:
      return []

File: development/bot_handler.py [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+from environment import get_env_variable
+
+from auth_app.models import Bot
+
+
+SERVER_URL = get_env_variable('SERVER_URL')
+SERVER_PORT = get_env_variable('SERVER_PORT')
+
+
+def get_data_users():
```
Comment (nicolasnunez-eb) [inline development/bot_handler.py:11]: NIT: I would rephrase it as `get_users_data`

File: development/tests.py [hunk 7]
```diff
@@ -42,3 +52,90 @@ def test_create_match(
...
+            self.json_data = json_data
+            self.status_code = status_code
+
+        def json(self):
+            return self.json_data
+
+    return MockResponse({key: data}, 200)
+
+
+class TestBotHandler(TestCase):
```
Comment (nicolasnunez-eb) [inline development/tests.py:72]: I think you are missing the scenario where you are catching an exception

**Review by nicolasnunez-eb — APPROVED** [review]
Nit comments, just mention that you are missing a test scenario where the request fails and you return empty users

File: development/bot_handler.py (hunk 5 above)
Comment (ibrizuela-eb) [reply development/bot_handler.py:21]: done!

File: development/tests.py (hunk 4 above)
Comment (ibrizuela-eb) [reply development/tests.py]: for the next sprint!

File: development/bot_handler.py (hunk 6 above)
Comment (ibrizuela-eb) [reply development/bot_handler.py:11]: done!

File: development/tests.py (hunk 7 above)
Comment (ibrizuela-eb) [reply development/tests.py:72]: for the next sprint!

File: development/forms.py (hunk 3 above)
Comment (ibrizuela-eb) [reply development/forms.py]: I need to explore and read documentation to fix this, but today is Friday (demo day) and I'm run out of time.
